### ISSUE TITLE
nginx with subpath reverse proxy issue with v10.4.1

### ISSUE THREAD

**Issue by jrbski** [jellyfin/jellyfin#1979]
**Describe the bug**

After installing Jellyfin v10.4.1 over 10.4.0, the nginx reverse proxy with a subpath of /jellyfin no longer works properly after removing  base URL path of /jellyfin

**To Reproduce**

1. Started with Jellyfin 10.4.0 with nginx reverse proxy on subpath of /jellyfin working properly
2. Downloaded and installed Jellyfin 10,4,1 and Jellyfin-ffmpeg updates
3. Use webui tp remove base URL path
4. Connect to https://server_ip:8096 - works properly
5. Connect to https://Public_DNS_Name/jellyfin/ - resolves to Public_DNS_Name/web/index.html#!/login.html and fails to load with '404 Not Found' from nginx
6. Workaround - Manually edit URL in Chrome to insert /jellyfin, i.e.- "Public_DNS_Name/jellyfin/web/index.html#!/login.html" - now works properly to get to login page, and subsequent URL's work properly

**Expected behavior**

https://Public_DNS_Name/jellyfin/ should resolve to jellyfin logn page

**Logs**

**Screenshots**

**System (please complete the following information):**
 - OS: Lubuntu 18.10 on AtomicPi
 - Browser: Chrome
 - Jellyfin Version: 10.4.1
 - Reverse proxy: nginx

**Additional context**

Subpath configuration on nginx was created using jellyfin documentation, and has been working fine since jellyfin 10.3. The initial URL may not be coded as relative (i.e.- "./web//index.html#!/login.html" as part of the 10.4.1 changes for base URL.

My nginx configuration details for the subpath are:

   location /jellyfin/ {
        # Proxy main Jellyfin traffic
        # The / at the end is significant.
        #resolver 127.0.0.1;
        # set $jellyserv 127.0.0.1:8096/;
        # proxy_pass http://$jellyserv/;
        proxy_pass http://127.0.0.1:8096/;
        #proxy_pass http://127.0.0.1:8096/jellyfin/;

        proxy_pass_request_headers on;
        proxy_set_header Host $host;

        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $http_host;

        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $http_connection;
    }

**Comment by nonplusnl** [comment]
I can confirm this exact issue (and workaround) also occurs when using Apache as a reverse proxy (Apache 2.4 on Ubuntu 18.04). So the bug is not a specific issue of Jellyfin interacting with nginx.

**Comment by anthonylavado** [comment]
Did you restart Jellyfin after changing the base url?

**Comment by jrbski** [comment]
Yes, restarted multiple times. I also tried changing nginx proxy_pass to /jellyfin/ with the base URL set, and while the webpages resolve, the livetv stream doesn't start (or is incorrect).

I had the best overall success removing the base url.

**Comment by nonplusnl** [comment]
Again a similar experience here. When using a base url other than /, streams to DLNA clients on the local network don't work.
As far as I can tell, these local DLNA clients are served directly by Jellyfin at http://serverIP:8096, and do not access Jellyfin via the reverse proxy (this is indeed desired behaviour; the proxy is there for https internet traffic; local network traffic can communicate over http directly). It appears that when Jellyfin is supposed to use http://serverIP:8096/<baseurl>/, something goes wrong serving local clients. 
So it appears we have two separate issues here:
1) Without base url set, there is a problem accessing Jellyfin via the proxy; 
2) With base url set, there is an issue serving local (DLNA/streaming) clients

**Comment by joshuaboniface** [comment]
DLNA should be using port 1900 and is handled separately from the main web server that is affected by the base URL, so that should not be related.

Clients should *not* be bypassing the reverse proxy in any scenario. We need to fix this on the client side.

**Comment by nonplusnl** [comment]
Sorry, I probably wasn't clear. The Web and Android clients are always routed via the reverse proxy. DLNA players are served directly by Jellyfin (bypassing the reverse proxy). In my case, DLNA clients include an Onkyo receiver and a local UpMpdCli DLNA front end to MPD. 
What I mentioned under '2' in my previous post, is that changing the BaseUrl to anything other than <BaseUrl /> breaks all local DLNA playback. As you indicate, this should not happen as DLNA is handled outside or the main web server. But still, changing BaseUrl does break DLNA playback in my case, so there appears to be some unintended effect at play here

**Comment by joshuaboniface** [comment]
That's OK, thanks for confirming! It indeed seems odd that DLNA is breaking with a base URL set, we'll have to investigate that in more detail.

**Comment by jrbski** [comment]
As a note, I also tested the latest FireTV app, and it does stream Livetv with the reverse proxy and no BaseUrl. I didn't try  FireTV with the a BaseUrl set.

**Comment by skug67** [comment]
Is the problem simply that there's a redirect or rewrite someplace taking you from "/" to "/web/index.html" that absolute instead of relative?  I tried searching the code for something like that and didn't find it.  But it sure seems like that must be what's going on.

**Comment by Sockolet** [comment]
I'm affecting with this in Apache as well, as you can see here:
https://www.reddit.com/r/jellyfin/comments/due27g/the_new_base_url_behavior_i_cant_connect_from/

Thanks @nonplusnl to bring me here.

**Comment by jrbski** [comment]
For some additional 'fun' I set-up a second instance of jellyfin in a docker container with a new nginx subpath of 'jellyfin3'. As I expected, I needed to use the workaround of "Public_DNS_Name/jellyfin3/web/index.html#!/login.html" to connect to the new server instead of the resolved URL of "Public_DNS_Name/web/index.html#!/login.html"

I was curious to see if the issue was that I was using the default base URL as my subpath. The check confirms that jellyfin appears to be using an absolute URL, not a relative URL.

**Comment by skug67** [comment]
Did a little digging.  Could it be line 18 of Emby.Server.Implementations/Browser/BrowserLauncher.cs

var url = appHost.GetLocalApiUrl("localhost") + "/web/" + page;

**Comment by skug67** [comment]
Did a little digging.  Could it be line 16 of Emby.Server.Implementations/Browser/BrowserLauncher.cs

var url = appHost.GetLocalApiUrl("localhost") + "/web/" + page;

**Comment by jrbski** [comment]
It looks like its only the referral to  the login page "web/index.html#!/login.html"... after that, the url's to other pages work fine.  My best guess right now it is related to the initial page referral to the login page,  but haven't tracked that down so far... may be in components/appRouter.js but stopped looking for the night!

**Comment by JustAMan** [comment]
If you wish to access Jellyfin with a baseURL, you should set it so (in the case of original submitter they should set baseURL to `/jellyfin` in server settings and strip the trailing slash in nginx config).

As for DLNA being broken when baseURL is anything non-empty, this is entirely separate issue being handled by a few PRs (https://github.com/jellyfin/jellyfin/pull/2020, https://github.com/jellyfin/jellyfin/pull/2019, https://github.com/jellyfin/jellyfin/pull/2008 IIRC)

**Comment by jrbski** [comment]
If I set the baseURL to anything other than blank as suggested by JustAMan, I am unable to stream liveTV. (No issues with a blank baseURL). Also, if I strip the trailing slash in nginx, nginx does not handle the reverse proxy properly. This may be a different bug, since I see the following in the transcode log (if baseURL is non-blank).  

As a note, I tested JustAMan's suggestion with a docker container, not the direct lubuntu install I reported on.

ffmpeg version 4.2.1 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.4.0-1ubuntu1~18.04.1)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-vaapi --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gnutls --enable-gpl --enable-libass --enable-libfreetype --enable-libvidstab --enable-libmp3lame --enable-libopenjpeg --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx265 --enable-libxvid --enable-libx264 --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-postproc --enable-cuvid --enable-nvenc --enable-version3 --enable-rpath --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib --extra-libs=-ldl --prefix=/opt/ffmpeg
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
http://127.0.0.1:8096/LiveTv/LiveStreamFiles/0cde249e4eec4c6d93d1ea1f058ea875/stream.ts: Invalid data found when processing input

**Comment by Sockolet** [comment]
Why was this closed? Is it solved?

**Comment by nonplusnl** [comment]
@jrbski I assume closing this was a mistake? The issues you identify are still present - indeed I experience the same issue of DLNA broken with BaseURL (similar to your LiveTV being broken), as I mentioned earlier

**Comment by jrbski** [comment]
Yes... not sure how I closed it!

Reopened - but I also see a new version was just released with some baseURL fixes, so I will try to update shortly

**Comment by jrbski** [comment]
OK.. I completed the update to 10.4.2 on my Lubuntu direct install, and there is no change with a blank baseURL. 

I'll update the docker image next, and see if the suggestions from JustAMan work with version 10.4.2.

Also, I now see how I closed this by clicking the wrong comment button :)

**Comment by jrbski** [comment]
Same results with docker v10.4.2 as 10.4.1... setting the baseURL to match the nginx proxy subpath fixes the login page issue, but LiveTV does not work. Using a blank baseURL breaks the login page, but using the work-around of manually adding the subpath to the URL allows login, and allows streaming of liveTV.

Same results for JustAMan's suggestion as I had for version 10.4.1

**Comment by JustAMan** [comment]
This looks to be a same issue as https://github.com/jellyfin/jellyfin/issues/2035, right?

**Comment by jrbski** [comment]
Yes, #2035 looks like a similar issue, If I set a baseURL to match the proxy subpath, I see a similar issue, although I am not using DLNA, so I get a different error message when the stream fails. Attached is the section of a log with a successful stream for comparison. No obvious hints, other than the message "[mpeg2video @ 0x56303f5ddd80] Invalid frame dimensions 0x0.
    Last message repeated 4 times" at the same spot that the other stream fails.

ffmpeg version 4.2.1 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.4.0-1ubuntu1~18.04.1)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-vaapi --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gnutls --enable-gpl --enable-libass --enable-libfreetype --enable-libvidstab --enable-libmp3lame --enable-libopenjpeg --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx265 --enable-libxvid --enable-libx264 --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-postproc --enable-cuvid --enable-nvenc --enable-version3 --enable-rpath --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib --extra-libs=-ldl --prefix=/opt/ffmpeg
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[mpeg2video @ 0x56303f5ddd80] Invalid frame dimensions 0x0.
    Last message repeated 4 times
Input #0, mpegts, from 'http://127.0.0.1:8096/LiveTv/LiveStreamFiles/f1a8cf84eb56471bafb8d43d50a4c507/stream.ts':
  Duration: N/A, start: 94530.921367, bitrate: N/A
  Program 1006 
    Stream #0:0[0x16db]: Video: mpeg2video (Main) ([2][0][0][0] / 0x0002), yuv420p(tv, top first), 704x480 [SAR 40:33 DAR 16:9], Closed Captions, 29.97 fps, 29.97 tbr, 90k tbn, 59.94 tbc
    Stream #0:1[0x16dc](eng): Audio: ac3 (AC-3 / 0x332D4341), 48000 Hz, 5.1(side), fltp, 384 kb/s
    Stream #0:2[0x16dd](spa): Audio: ac3 (AC-3 / 0x332D4341), 48000 Hz, stereo, fltp, 192 kb/s
Only '-vf yadif=0:-1:0,scale=trunc(min(max(iw\,ih*dar)\,1920)/2)*2:trunc(ow/dar/2)*2' read, ignoring remaining -vf options: Use ',' to separate filters
Only '-af volume=2' read, ignoring remaining -af options: Use ',' to separate filters
Stream mapping:
  Stream #0:0 -> #0:0 (mpeg2video (native) -> h264 (libx264))
  Stream #0:1 -> #0:1 (ac3 (native) -> mp3 (libmp3lame))
Press [q] to stop, [?] for help
[libx264 @ 0x56303f617f00] using SAR=1912/1917
[libx264 @ 0x56303f617f00] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2
[libx264 @ 0x56303f617f00] profile Main, level 4.1
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b660.ts' for writing
Output #0, segment, to '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b66%d.ts':
  Metadata:
    encoder         : Lavf58.29.100
    Stream #0:0: Video: h264 (libx264), yuv420p, 852x478 [SAR 1912:1917 DAR 16:9], q=-1--1, 29.97 fps, 90k tbn, 29.97 tbc
    Metadata:
      encoder         : Lavc58.54.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 5000000/0/0 buffer size: 10000000 vbv_delay: -1
    Stream #0:1: Audio: mp3 (libmp3lame), 48000 Hz, stereo, fltp, 384 kb/s
    Metadata:
      encoder         : Lavc58.54.100 libmp3lame
frame=   41 fps=0.0 q=29.0 size=N/A time=00:00:00.69 bitrate=N/A dup=22 drop=0 speed=1.38x    
frame=   68 fps= 67 q=29.0 size=N/A time=00:00:01.65 bitrate=N/A dup=22 drop=0 speed=1.64x    
frame=   95 fps= 63 q=29.0 size=N/A time=00:00:02.61 bitrate=N/A dup=22 drop=0 speed=1.73x    
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b66.m3u8.tmp' for writing
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b661.ts' for writing
frame=  122 fps= 60 q=29.0 size=N/A time=00:00:03.57 bitrate=N/A dup=22 drop=0 speed=1.77x    
frame=  152 fps= 60 q=29.0 size=N/A time=00:00:04.36 bitrate=N/A dup=22 drop=0 speed=1.73x    
frame=  176 fps= 58 q=29.0 size=N/A time=00:00:05.32 bitrate=N/A dup=22 drop=0 speed=1.76x    
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b66.m3u8.tmp' for writing
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b662.ts' for writing
frame=  203 fps= 57 q=29.0 size=N/A time=00:00:06.21 bitrate=N/A dup=22 drop=0 speed=1.76x    
frame=  227 fps= 56 q=29.0 size=N/A time=00:00:07.03 bitrate=N/A dup=22 drop=0 speed=1.74x    
frame=  244 fps= 54 q=29.0 size=N/A time=00:00:07.56 bitrate=N/A dup=22 drop=0 speed=1.66x    
frame=  261 fps= 52 q=29.0 size=N/A time=00:00:08.04 bitrate=N/A dup=22 drop=0 speed=1.59x    
frame=  272 fps= 48 q=29.0 size=N/A time=00:00:08.59 bitrate=N/A dup=22 drop=0 speed=1.53x    
frame=  290 fps= 47 q=29.0 size=N/A time=00:00:09.14 bitrate=N/A dup=22 drop=0 speed= 1.5x    
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b66.m3u8.tmp' for writing
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b663.ts' for writing
[segment @ 0x56303f5d4a00] Opening '/config/transcoding-temp/transcodes/transcodes/transcodes/537f2de8757be2d4787e73455f346b66.m3u8.tmp' for writing
frame=  292 fps= 47 q=-1.0 Lsize=N/A time=00:00:09.64 bitrate=N/A dup=22 drop=0 speed=1.54x    
video:185kB audio:362kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
[libx264 @ 0x56303f617f00] frame I:4     Avg QP:15.95  size:  9632
[libx264 @ 0x56303f617f00] frame P:82    Avg QP:18.66  size:  1292
[libx264 @ 0x56303f617f00] frame B:206   Avg QP:21.83  size:   218
[libx264 @ 0x56303f617f00] consecutive B-frames:  3.1%  7.5%  3.1% 86.3%
[libx264 @ 0x56303f617f00] mb I  I16..4: 79.4%  0.0% 20.6%
[libx264 @ 0x56303f617f00] mb P  I16..4: 11.0%  0.0%  0.0%  P16..4: 16.5%  0.0%  0.0%  0.0%  0.0%    skip:72.5%
[libx264 @ 0x56303f617f00] mb B  I16..4:  0.3%  0.0%  0.0%  B16..8:  1.0%  0.0%  0.0%  direct: 2.9%  skip:95.8%  L0:33.1% L1:66.8% BI: 0.2%
[libx264 @ 0x56303f617f00] coded y,uvDC,uvAC intra: 17.5% 38.5% 4.8% inter: 0.6% 5.4% 0.0%
[libx264 @ 0x56303f617f00] i16 v,h,dc,p: 39% 28% 19% 14%
[libx264 @ 0x56303f617f00] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 30% 25% 26%  5%  5%  3%  2%  2%  2%
[libx264 @ 0x56303f617f00] i8c dc,h,v,p: 55% 23% 18%  3%
[libx264 @ 0x56303f617f00] Weighted P-Frames: Y:0.0% UV:0.0%
[libx264 @ 0x56303f617f00] kb/s:155.50

**Comment by JustAMan** [comment]
@Bond-009 @joshuaboniface ideas?

**Comment by Sockolet** [comment]
Any update on this? It's still broken in new 1.4.2...

**Comment by JustAMan** [comment]
Define "it" please :)

**Comment by jrbski** [comment]
I updated to 10.4.3, and I am still seeing the same 'catch22'. If base URL is blank, I need to perform the work-around of entering the nginx subpath on the first redirect, and after that, all works as expected. If I set a base URL, the reverse proxy works fine, but liveTV no longer works. I did test the liveTV with a baseURL set without using the reverse proxy (directly to the server), and it also fails to stream liveTV.

There appears to be 2 separate issues. Leaving the baseURL blank, and manually fixing the first redirect is the best work-around. I am also having no issues running the FireTV app with this configuration.

**Comment by TraumaGuy** [comment]
Any updates on this issue? as previously stated, It is still present in 10.4.3

I can also confirm that the workaround works fine, and that using subdomains linked to root as location (i.e. http://jellyfin.mydomain.com) also works perfectly fine.

**Comment by neuroid** [comment]
FWIW, adding these two lines in the location block should solve the "first redirect" issue, without the need to set `Base URL`:

```
proxy_redirect default;
proxy_redirect / /jellyfin/;
```

Tested on 10.4.3 only.

**Comment by jrbski** [comment]
I just tested the nginx config update suggested by neuroid using my docker config under 10.4.3, and can confirm this works if you add the proxy_redirect that matches the subpath. (My docker reverse subpath was /jellyfin3/ ). LiveTV works properly

This simple nginx config change solves the issue if you have a blank Base URL, so it solved the  problem for me, as this was my current set-up.

Thanks neuroid!

**Comment by TraumaGuy** [comment]
I'll be testing this solution as soon as possible.

If this really works, perhaps it should be included in the Jellyfin documentation as base code needed for it to work.

I'll report back.

**Comment by TraumaGuy** [comment]
The workaround provided by Neuroid indeed works.

Thanks!

**Comment by jrbski** [comment]
Agreed... it looks like an update to the nginx instructions to include the re-direct may be the 'permanent fix' for the issue with the baseURL blank. I did see some updates to correct handling of a non-blank baseURL, so it will be interesting to retest on the next release.

**Comment by Sockolet** [comment]
Yep, tried it with my new nginx configuration and seems to be a reliable workaround. Working again right now!

**Comment by JustAMan** [comment]
Should be fixed by https://github.com/jellyfin/jellyfin/pull/2221

**Comment by RaymondSchnyder** [comment]
> 
> 
> Should be fixed by #2221

I have built 10.5.0, use a baseurl and use a common nginx reverse config. 
Im stuck at the loading screen, as it connect to jellyfin
Ive saw logs earlier like "couldnt connect to /jellyfin/jellyfin/authenticatebyname" which is weird, because now it wont log anything related to that.

tldr: either this issue still persists or im dumb.

**Comment by JustAMan** [comment]
@RaymondSchnyder can you post relevant nginx config here?

**Comment by RaymondSchnyder** [comment]
> 
> 
> @RaymondSchnyder can you post relevant nginx config here?


    location /jellyfin/ {
        error_log /var/log/nginx/jellyfin_error.log;
        access_log /var/log/nginx/jellyfin_access.log;
        # Proxy main Jellyfin traffic
        proxy_pass http://127.0.0.1:8096/jellyfin/;

        proxy_pass_request_headers on;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Protocol $scheme;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $http_connection;
    }

BaseURL is configured as "/jellyfin"

Theres no error in the jellyfin log (even with debug enabled).
NGINX access log says that:
`"POST /jellyfin/Users/authenticatebyname HTTP/1.0" 400 0 "-" `

**Comment by JustAMan** [comment]
That error is a separate issue, might be already fixed in master.

**Comment by RaymondSchnyder** [comment]
> 
> 
> That error is a separate issue, might be already fixed in master.
I agree, i will build the current master and if it happens again, i will create a new issue.

**Comment by skaa** [comment]
Same stacktrace while logging with wrong password:
`[2020-03-27 08:43:29.053 +05:00] [INF] Authentication request for "admin" "has been denied".
[2020-03-27 08:43:29.054 +05:00] [ERR] Error processing request
System.NullReferenceException: Object reference not set to an instance of an object.
   at Emby.Server.Implementations.Session.SessionManager.GetAuthorizationToken(User user, String deviceId, String app, String appVersion, String deviceName)
   at Emby.Server.Implementations.Session.SessionManager.AuthenticateNewSessionInternal(AuthenticationRequest request, Boolean enforcePassword)
   at MediaBrowser.Api.UserService.Post(AuthenticateUserByName request)
   at Emby.Server.Implementations.Services.ServiceExecGeneral.GetTaskResult(Task task)
   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken)
   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)`

- OS: Debian 10 (OpenMediaVault 5.3.8-1)
- Browser: Firefox
- Jellyfin Version: 10.5.2
- Reverse proxy: not used

**Comment by JustAMan** [comment]
cc @joshuaboniface @Bond-009 for ideas

**Comment by AlexandreBonneau** [comment]
For info, the bug (and the workaround) is also happening on Debian with the official jellyfin package `10.6.4-1`.

When you go to `http://localname:8096/` you see the Jellyfin logo briefly, then a black screen and get stuck there.
`localname` here is set in /etc/hosts as an alias for 10.0.0.1, where the server hosting jellyfin is.
In the log, you see:
```
 [ERR] Error processing request. URL: "http://localname:8096/users/public"
System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.ThrowHelper.ThrowNoMatchException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
   at Jellyfin.Data.Entities.User.GetPreference(PreferenceKind preference)
   at Jellyfin.Server.Implementations.Users.UserManager.GetUserDto(User user, String remoteEndPoint)
   at MediaBrowser.Api.UserService.<>c__DisplayClass8_0.<Get>b__4(User i)
   at System.Linq.Enumerable.SelectIPartitionIterator`2.LazyToArray()
   at MediaBrowser.Api.UserService.Get(GetUsers request, Boolean filterByDevice, Boolean filterByNetwork)
   at MediaBrowser.Api.UserService.Get(GetPublicUsers request)
   at Emby.Server.Implementations.Services.ServiceExecGeneral.Execute(Type serviceType, IRequest request, Object instance, Object requestDto, String requestName)
   at Emby.Server.Implementations.Services.ServiceController.Execute(HttpListenerHost httpHost, Object requestDto, IRequest req)
   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken)
   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)
```

If you manually edit the url to add `web/index.html#!/login.html` then it shows you the correct login page.
This bug breaks the `jellyfin-mpv-shim` application too.

**Comment by HybridSarcasm** [comment]
This has been open quite a while. Can this be re-produced with the current build? If so, please update the bug report details. Thanks!
